pkg/registry/registry.go

```diff
@@ -269,11 +388,20 @@ func createRegistryDevfiles(registry api.Registry, devfileIndex []indexSchema.Sc
 func (o RegistryClient) retrieveDevfileDataFromRegistry(ctx context.Context, registryName string, devfileName string) (api.DevfileData, error) {
 
 	// Create random temporary file
-	tmpFile, err := os.MkdirTemp("", "odo")
+	tmpFile, err := o.fsys.TempDir("", "odo")
 	if err != nil {
 		return api.DevfileData{}, err
 	}
-	defer os.Remove(tmpFile)
+	err = o.fsys.MkdirAll(tmpFile, 0700)
```

Comment:
I think `o.fsys.TempDir` should create the temporary directory, so not sure we need this extra call.